scripts/sct_deepseg.py

```diff
@@ -0,0 +1,169 @@
+#!/usr/bin/env python
+# -*- coding: utf-8
+"""
+This command-line tool is the interface for the deepseg API that performs segmentation using deep learning from the
+ivadomed package.
+"""
+
+# TODO: Add link to example image so users can decide wether their images look "close enought" to some of the proposed
+#  models (e.g., mice, etc.).
+# TODO: add test to make sure that all postprocessing flags match the core.DEFAULT dictionary items
+# TODO: Fetch default value (and display) depending on the model that is used.
+
+from __future__ import absolute_import
+
+import sys
+import os
+import argparse
+import colored
+
+import spinalcordtoolbox as sct
+import spinalcordtoolbox.deepseg.core
+import spinalcordtoolbox.deepseg.models
+import spinalcordtoolbox.utils
+
+from sct_utils import init_sct, printv, display_viewer_syntax
+
+
+def get_parser():
+
+    parser = argparse.ArgumentParser(
+        description="Segmentation using deep learning.",
```

Comment:
This section should mention that multiclass segmentation is a feature. Or maybe there should be a flag for doing multiclass segmentation? Or the models metadata should include if it does multiclass segmentation? The user should be able to discover this feature very quickly because it could cause a lot of confusion if they weren't expecting it. The comment below about " In case of multi-class segmentation, class-specific suffixes will be added" was a surprise to me.

It doesn't really matter that this uses neural network models

Reply:
A broader comment: it doesn't really matter that this uses deep learning. I would probably not make UI that exposes that specifically to the user. The important user-facing thing is that it does segmentation. Is there some detail about the deep learning models that users need to be aware of (maybe `-thr`)?

What I would rather do is:
- separate all the deep learning and non-deep learning models out into subpackages
- make them all optional dependencies using [extras] (https://github.com/neuropoly/spinalcordtoolbox/issues/2712#issuecomment-629703499)
- provide a single `sct_segment $model $image` interface; if the model is a deeplearning model, then jump to `spinalcordtoolbox.deepseg`; if it's another kind of segmentation model, go to the other relevant part of the code.

That's a pretty big change to ask for now so I'm not asking for it, just highlighting that the approach is out there.

Reply:
> This section should mention that multiclass segmentation is a feature. Or maybe there should be a flag for doing multiclass segmentation? 

i removed multiclass from the usage 490578cd32688d7f7fb0550958b05c7efa3c1443

> separate all the deep learning and non-deep learning models out into subpackages

I don't understand this comment;  all models are based on deep learning

> make them all optional dependencies using [extras] (#2712 (comment))
provide a single sct_segment $model $image interface; if the model is a deeplearning model, then jump to spinalcordtoolbox.deepseg; if it's another kind of segmentation model, go to the other relevant part of the code.
That's a pretty big change to ask for now so I'm not asking for it, just highlighting that the approach is out there.

Indeed, this is big. Awesome, but big. I don't have the time to do it.